### PR TITLE
Improve generation of function template variable names

### PR DESCRIPTION
**Pull request by zth**
We're able to generate function template/snippets for callbacks since a while back. But, given that the type information has no name for unlabelled arguments, we've been forced to call most variables in the generated template `v` (with a few notable exceptions that we've special cased). 

This PR improves how we infer template variable names by looking at the type path. For example, the type `Environment.t` would be called `environment` in templates, whereas `Environment.envConfig` would be called `envConfig`.

This can help the DX quite a lot, because we can capture _some_ of the variable name intent, at least for more complex types.

Before:
image

After:
image

I think this will be quite useful, but interested in hearing more thoughts.

### CODE REVIEW

**Comment by cristianoc** [comment]
Looks like CI is red.

**Comment by zth** [comment]
No idea why CI is red now, everything passes locally.

File: analysis/tests/src/CompletionExpressions.res [hunk 1]
```diff
@@ -196,3 +196,59 @@ ignore(fff)
 
 // fff.someOpt
 //            ^com
+
+@editor.templateVariableName("someTyp")
```
Comment (cristianoc) [inline analysis/tests/src/CompletionExpressions.res:200]: This annotation does not seem to be doing anything?

Comment (cristianoc) [reply analysis/tests/src/CompletionExpressions.res:200]: In that if you comment it out, tests don't change.

**Comment by cristianoc** [comment]
Not sure one can turn `Environment.t` into `environment`. Could it be `Environment`? Or `Environment_t`? Just so it's clear it does not refer to something else.

**Review by cristianoc — APPROVED** [review]

**Comment by zth** [comment]
@cristianoc I removed the custom attribute stuff, and just went with improving the template variable names for now.

**Review by cristianoc — APPROVED** [review]

**Pull request merged**